pymc3/util.py

```diff
@@ -24,7 +24,20 @@
 
 from cachetools import LRUCache, cachedmethod
 
-UNSET = object()
+
+class _UnsetPlaceholder:
+    """Instances of this type can be used as default values for (keyword) arguments
+    for which `None` is supposed to result in a different behavior than not passing
+    the argument at all."""
+
+    def __str__(self):
+        return "not set"
+
+    def __repr__(self):
+        return str(self)
+
+
+UNSET = _UnsetPlaceholder()
```

Comment:
What is the point of this?  Does a simple sentinel value really need a `__str__`, `__repr__`, and/or a custom type?

Reply:
Not having a concise str representation makes tracebacks, error messages and warnings much longer and more confusing.
I added this right at the start when working on this branch, because it was so annoying..

Reply:
Can you give an example of where/when this arises?

Reply:
The pytest traceback logs incoming variables:
`cls = <class 'pymc3.tests.test_initvals.NormalWithoutInitval'>, dist_params = [TensorConstant{1.0}, TensorConstant{2.0}], shape = None, size = None, initval = not set
kwargs = {'rng': RandomStateSharedVariable(<RandomState(MT19937) at 0x2145DA58640>)}
`
vs.
`
cls = <class 'pymc3.tests.test_initvals.NormalWithoutInitval'>, dist_params = [TensorConstant{1.0}, TensorConstant{2.0}], shape = None, size = None
initval = <pymc3.util._UnsetPlaceholder object at 0x000002017CC29130>, kwargs = {'rng': RandomStateSharedVariable(<RandomState(MT19937) at 0x2017EA98640>)}
`

I needed the type anyways to do `isinstance` checks.